src/boolean3.cpp

```diff
@@ -573,6 +580,11 @@ Boolean3::Boolean3(const Manifold::Impl &inP, const Manifold::Impl &inQ,
       Intersect12(inQ, inP, s20, p2q0, s11, p1q1, z20, xyzz11, p2q1_, false);
   PRINT("x21 size = " << x21_.size());
 
+  if (x12_.size() > INT_MAX_SZ || x21_.size() > INT_MAX_SZ) {
```

Comment:
x12 and x21 can be at most the size of the sparse indices, which is at most n^2 and can cause overflow elsewhere.